.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,27 @@
+on: [push, pull_request]
```

Comment:
Maybe add a short comment explaining what this workflow does and why (also where it's been taken from) :)